contracts/colonyNetwork/ColonyNetworkDeployer.sol

```diff
@@ -175,6 +178,94 @@ contract ColonyNetworkDeployer is ColonyNetworkStorage {
     // This is intentional, as we want to allow the same Colony to be deployed on different chains
   }
 
+  function setDomainTokenReceiverResolver(address _resolver) public stoppable auth {
+    domainReceiverResolverAddress = _resolver;
+  }
+
+  function getDomainTokenReceiverResolver() public view returns (address) {
+    return domainReceiverResolverAddress;
+  }
+
+  function checkDomainTokenReceiverDeployed(
+    uint256 _domainId
+  ) public stoppable calledByColony returns (address domainTokenReceiverAddress) {
+    // Calculate the address the domain should be receiving funds at
+    domainTokenReceiverAddress = getDomainTokenReceiverAddress(msgSender(), _domainId);
+
+    if (!isContract(domainTokenReceiverAddress)) {
+      // Then deploy the contract
+      bytes32 salt = getDomainTokenReceiverDeploySalt(msgSender(), _domainId);
+      address deployedAddress = ICreateX(CREATEX_ADDRESS).deployCreate3AndInit(
```

Comment:
Can we pull this into a helper function which only takes the salt? I've seen this code repeated a few times.